### PR TITLE
[Merged by Bors] - feat(NumberTheory/LSeries/PrimesInAP): von Mangoldt restricted to a residue class

### PR DESCRIPTION
**Pull request by MichaelStollBayreuth**
This is the next step on the way to **Dirichlet's Theorem**. It defines `ArithmeticFunction.vonMangoldt.residueClass` as the function that is the von Mangoldt function on a given residue class and zero outside and proves some facts about it (in particular that it is a linear combination of twists of the von Mangoldt function by Dirichlet characters).

See [here](https://leanprover.zulipchat.com/#narrow/channel/144837-PR-reviews/topic/Prerequisites.20for.20PNT.20and.20Dirichlet's.20Thm/near/483785853) on Zulip.

The large import increase is caused by the need for more material to be able to state and prove the new things.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/NumberTheory/LSeries/PrimesInAP.lean [hunk 1]
```diff
@@ -65,3 +66,82 @@ lemma tprod_eq_tprod_primes_mul_tprod_primes_of_mulSupport_subset_prime_powers {
...
+namespace ArithmeticFunction.vonMangoldt
+
+open Complex LSeries DirichletCharacter
+
+open scoped LSeries.notation
+
+variable {q : ℕ} (a : ZMod q)
+
+/-- The von Mangoldt function restricted to the residue class `a` mod `q`. -/
+noncomputable abbrev residue_class : ℕ → ℝ :=
```
Comment (alreadydone) [inline Mathlib/NumberTheory/LSeries/PrimesInAP.lean:85]:
```suggestion
noncomputable abbrev residueClass : ℕ → ℝ :=
```

Comment (MichaelStollBayreuth) [reply Mathlib/NumberTheory/LSeries/PrimesInAP.lean]: Thanks; I always stumble over this...

File: Mathlib/NumberTheory/LSeries/PrimesInAP.lean [hunk 2]
```diff
@@ -65,3 +66,85 @@ lemma tprod_eq_tprod_primes_mul_tprod_primes_of_mulSupport_subset_prime_powers {
...
+noncomputable abbrev residueClass : ℕ → ℝ :=
+  {n : ℕ | (n : ZMod q) = a}.indicator (vonMangoldt ·)
+
+lemma residueClass_nonneg (n : ℕ) : 0 ≤ residueClass a n :=
+  Set.indicator_apply_nonneg fun _ ↦ vonMangoldt_nonneg
+
+lemma residueClass_le (n : ℕ) : residueClass a n ≤ vonMangoldt n :=
+  Set.indicator_apply_le' (fun _ ↦ le_rfl) (fun _ ↦ vonMangoldt_nonneg)
+
+lemma residueClass_apply_zero : residueClass a 0 = 0 := by
```
Comment (jcommelin) [inline Mathlib/NumberTheory/LSeries/PrimesInAP.lean:94]: Is this worth being a simp lemma?
```suggestion
@[simp]
lemma residueClass_apply_zero : residueClass a 0 = 0 := by
```

File: Mathlib/NumberTheory/LSeries/PrimesInAP.lean [hunk 3]
```diff
@@ -65,3 +66,85 @@ lemma tprod_eq_tprod_primes_mul_tprod_primes_of_mulSupport_subset_prime_powers {
...
+/-- We can express `ArithmeticFunction.vonMangoldt.residueClass` as a linear combination
+of twists of the von Mangoldt function with Dirichlet charaters. -/
+lemma residueClass_eq (ha : IsUnit a) :
+    ↗(residueClass a) = (q.totient : ℂ)⁻¹ •
+      ∑ χ : DirichletCharacter ℂ q, χ a⁻¹ • (fun n : ℕ ↦ χ n * vonMangoldt n) := by
+  ext1 n
+  simpa only [Pi.smul_apply, Finset.sum_apply, smul_eq_mul, ← mul_assoc]
+    using residueClass_apply ha n
+
+/-- The L-series of the von Mangoldt function restricted to the prime residue class `a` mod `q`
```
Comment (jcommelin) [inline Mathlib/NumberTheory/LSeries/PrimesInAP.lean:132]: There is no assumption that `q` is prime, right?

Comment (MichaelStollBayreuth) [reply Mathlib/NumberTheory/LSeries/PrimesInAP.lean:132]: "Prime" in the sense of "prime to `q`". I'll clarify.

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**